pkg/kubelet/dockertools/manager.go

```diff
@@ -2052,6 +2050,9 @@ func (dm *DockerManager) GetPodStatus(uid types.UID, name, namespace string) (*k
 		}
 		containerStatuses = append(containerStatuses, result)
 		if ip != "" {
+			if podStatus.IP != ip {
```

Comment:
Add some comments on ordering here @pmorie! We don't want a refactoring to screw up this logic later.